### PR TITLE
chore(similarity): Enable remaining platforms

### PR DESCRIPTION
**Pull request by armenzg**
This adds the remaining platform for AI grouping with a few exceptions: the platform is not set, it is `other` or `native*`.

These platform exceptions are because we are not certain of the outcome and require a bit more investigation.

### CODE REVIEW

File: src/sentry/seer/similarity/utils.py [hunk 1]
```diff
@@ -159,6 +159,45 @@
         "java-spring",
         "java-spring-boot",
         "perl",
+        # Remaining platforms
+        "apple",
+        "apple-ios-profiling-onboarding-1-install",
+        "apple-ios-profiling-onboarding-4-upload",
+        "apple-macos",
+        "c",
```
Comment (lobsterkatie) [inline src/sentry/seer/similarity/utils.py:167]: Is c not native? Same for minidump.

File: src/sentry/seer/similarity/utils.py [hunk 2]
```diff
@@ -159,6 +159,45 @@
...
+        "javascript-nextjs",
+        "javascript-nuxt",
+        "javascript-solidstart",
+        "kotlin",
+        "minidump",
+        # "native", Holding off for now
+        # "native-qt",
+        "nintendo-switch",
+        "objc",
+        # "other" and null platform will be excluded for now
```
Comment (lobsterkatie) [inline src/sentry/seer/similarity/utils.py:194]: We need to exclude the empty string, too, as well as `null`. Seems like the code does do that, but we should probably include at least a mention of it here.

**Review by lobsterkatie — COMMENTED** [review]
A few comments on specific entries, but I think might (?) be a disconnect here, because our naming is bad. 

The platforms are being added to `SEER_ELIGIBLE_PLATFORMS`, but that's not what's checked in the function where you added the metric. There we're checking `SEER_ELIGIBLE_PLATFORMS_EVENTS`.

Was this intentional?

(Also, at this point, might a blocklist be a better choice rather than listing out all of the gross variations which show up in `project.platform`?)

File: src/sentry/seer/similarity/utils.py (hunk 1 above)
Comment (armenzg) [reply src/sentry/seer/similarity/utils.py:167]: I honestly don't know. I will comment it out.

Comment (lobsterkatie) [reply src/sentry/seer/similarity/utils.py:167]: Okay, sorry. I should be less circumspect. 😛  C and minidump are native, and we shouldn't handle them yet.

File: src/sentry/seer/similarity/utils.py [hunk 3]
```diff
@@ -17,12 +17,23 @@
 FULLY_MINIFIED_STACKTRACE_MAX_FRAME_COUNT = 20
 SEER_ELIGIBLE_PLATFORMS_EVENTS = frozenset(
     [
-        "csharp",
```
Comment (armenzg) [inline src/sentry/seer/similarity/utils.py:20]: @lobsterkatie pointed me to this short list:
https://github.com/getsentry/relay/blob/20f72be4c573355ea0374bcaf5a3ff3fbabe29d5/relay-event-schema/src/protocol/constants.rs#L2C11-L22

**Review by lobsterkatie — APPROVED** [review]
LGTM!

**Pull request merged**